Source/missiles.cpp

```diff
@@ -2766,17 +2691,20 @@ void AddDiabApoca(Missile &missile, Point /*dst*/, Direction /*midir*/)
 	missile._miDelFlag = true;
 }
 
-int AddMissile(Point src, Point dst, Direction midir, missile_id mitype, mienemy_type micaster, int id, int midam, int spllvl)
+int AddMissile(Point src, Point dst, Direction midir, missile_id mitype, mienemy_type micaster, int id, int midam, int spllvl, Missile *pParent /*= nullptr*/)
 {
-	if (ActiveMissileCount >= MAXMISSILES - 1)
-		return -1;
+	constexpr int32_t MaxMissiles = std::numeric_limits<int32_t>::max();
 
-	int mi = AvailableMissiles[0];
-	auto &missile = Missiles[mi];
+	int32_t mi = 0;
+	while (true) {
+		if (Missiles.find(mi) == Missiles.end())
+			break;

```

Comment:
Isn't this the same?
```suggestion
	while (Missiles.find(mi) != Missiles.end())
```

Reply:
We could probably get rid of this loop if we track the lowest available number (increase by 1 each time `.insert()` is called and `global = min(key, global)` each time `.erase()` is called).

Reply:
I changed the loop. 🙂 

For getting rid of it. I think we can remember where we can start the loop. But don't get rid of it.
Example:
Missiles 1,2,3 exists.
Missile 1 gets deleted.
global = 1

Now we insert two Missiles
Missile 1
and after that the next would be Missile 4.
But to get the index 4 we need the loop.

Reply:
Good point, it's probably just complicating things more then it's helping so lets leave it as is.